src/Stratis.Bitcoin.Features.MemoryPool/MempoolRuleContext.cs

```diff
@@ -0,0 +1,35 @@
+﻿using Microsoft.Extensions.Logging;
+using NBitcoin;
+using Stratis.Bitcoin.Consensus;
+using Stratis.Bitcoin.Features.MemoryPool.Interfaces;
+
+namespace Stratis.Bitcoin.Features.MemoryPool
+{
+    public class MempoolRuleContext
```

Comment:
Add a description for this class - e.g. the characteristics that make it different from `MempoolValidationContext`.

Reply:
Would making the rules singletons make this class unnecessary? It seems we are guessing the information future rules may require when we could use DI instead.